### ISSUE TITLE
We can remove cert-manager this depend

### ISSUE THREAD

**Issue by lengrongfu** [goharbor/harbor-operator#1085]
Current `harbor-operator` depends on too many components, we hope can remove `cert-manager` this depend.

**Comment by lengrongfu** [comment]
My goal is to be as simple and easy to use as `harbor-helm`.

**Comment by Vad1mo** [comment]
makes sense, however it shhould also still be self contained. How do you plan to make it work instead?

**Comment by lengrongfu** [comment]
Sorry, I haven't decided what needs to be done yet, but I'm considering taking this opportunity to improve the project. My goal is to enable users to deploy a harbor successfully without relying on other components.

**Comment by Vad1mo** [comment]
Makes sense. 

Before doing something that nobody want I would suggest you outline how the new solution will look and work for the user. Maybe a happy path installing and upgrading 

This easy we have a foundation for discussion and can see if this makes sense to be part of the operator

**Comment by lengrongfu** [comment]
More precisely, I hope `cert-manager`  to be optional.

There are two uses of cert-manager in the current project:
1. Add the annotation `cert-manager.io/inject-ca-from: harbor-operator-ns/serving-cert` in Webhook to configure the HTTPS certificate.
2. When the harbor components core, job, and register directly enable HTTPS communication, a certificate needs to be created, but this is optional.


So I want to change the first point to an optional operation and use a project similar to [kube-webhook-certgen](https://github.com/jet/kube-webhook-certgen) to generate https certificates.

**Comment by lengrongfu** [comment]
@Vad1mo what do you think?

**Comment by Vad1mo** [comment]
If it is difficult to explain in text you can join the community meeting and outline your idea, so we can discuss and make a decission
